### ISSUE TITLE
aria-describedby reports referencing error, for empty spans that will have error messages.

### ISSUE THREAD

**Issue by DavidMacDonald** [dequelabs/axe-core#1151]
A site has gets injected with a span that is populated with an error message if the user submits the form without filling in the field that is required. Axe reporting an aria-describedby  error. 

Invalid ARIA attribute value: aria-describedby="giftCard_0Error"

```<input id="giftCard_0" aria-describedby="giftCard_0Error" aria-invalid="false" ...```

Because there is no corresponding element with a matching id until the user makes an error and submits.
does it require an empty span?
```<span id="giftCard_0Error" aria-live="polite">[nothing here yet]></span>```

I wonder if there is some way to make  the aria-describedby message conditional to the presence of an error message?? Not sure, but its a false positive as it stands.

**Comment by DavidMacDonald** [comment]
I think its valuable to check that there is a matching id. Its fine if its a separate check. However, I think perhaps downgrading the check for corresponding id if the aria-describedby on a input type="text" to needs manual conformation.

**Comment by jeankaplansky** [comment]
@WilcoFiers ,  @straker : Please confirm - no new documentation required here until we create a new rule specifically testing the IDREF? Thanks.

**Comment by straker** [comment]
@jeankaplansky Correct. No new documentation required.

**Comment by jeankaplansky** [comment]
👍 Thanks!

**Comment by chandana7393** [comment]
Tested, under "We are not sure this is an issue, because:" observed that for ARIA attribute unable to see the value instead some condition is displayed.
![checking_id_ref](https://user-images.githubusercontent.com/45285854/62190619-6bb17d00-b38f-11e9-8273-9ffab339b60f.PNG)
**Tested Environment:**
Attest - 2.5.1.21115v
Axe - 3.8.1.21115v
Axe-coconut - 3.8.1.21104v
Chrome - 75.0.3770.142v
OS - Windows 10 64 bit

**Comment by somaalapati** [comment]
@chandana7393 I hope you can verify it now.

**Comment by chandana7393** [comment]
Tested, working fine on axe-coconut.
![ref_id](https://user-images.githubusercontent.com/45285854/62513381-8aa18a80-b839-11e9-9e12-7d641a035eae.PNG)
For Attest and axe under "We are not sure this is an issue, because:" observed that for ARIA attribute unable to see the value instead some condition is displayed.
![axe_id_ref](https://user-images.githubusercontent.com/45285854/62513407-9f7e1e00-b839-11e9-8b60-0c49ee279fdb.PNG)
![Attest_id_ref](https://user-images.githubusercontent.com/45285854/62513415-a3aa3b80-b839-11e9-9fdb-4358261017b4.PNG)
**Tested Environment:**
Attest - 2.5.1.21421v
Axe - 3.8.1.21421v
Axe-coconut - 3.8.1.21704v
Chrome - 76.0.3809.87v
Firefox - 68.0.1v
OS - Windows 10 64 bit.

**Comment by straker** [comment]
@WilcoFiers we need to create another release of axe-core to fix the incomplete translation message for all downstream dependencies.

**Comment by straker** [comment]
@chandana7393 since this requires a new version of axe and Attest to pull it in, could we close this issue as the problem has been resolved?

**Comment by chandana7393** [comment]
Tested, working as expected in axe-coconut.
![check_ref_axe_coconut](https://user-images.githubusercontent.com/45285854/62767015-5e903e80-bab1-11e9-916f-ab82be163acc.PNG
**Tested Environment:**
Axe-coconut - 3.8.1.21704v
Chrome - 76.0.3809.100v
OS - Windows 10 64 bit.
Created new ticket for Axe and Attest
https://github.com/dequelabs/axe-core/issues/1761

**Comment by iamrafan** [comment]
@WilcoFiers @straker : The intended fix for this issue is not working as expected. Case-1 in the below samples does not show up as "needs review'.

Demo: https://codepen.io/iamrafan/pen/abOrzVb?editors=1000

```aria-describedby``` referenced to an empty ```<span>``` is now not a failure but it does not show up as 'needs review' as well.
